dash/include/dash/halo/HaloMatrixWrapper.h

```diff
@@ -200,104 +217,140 @@ class HaloMatrixWrapper {
+   * The \ref GlobalBoundarySpec is set to default.
+   */
+
+  template <typename... StencilSpecT>
+  HaloMatrixWrapper(MatrixT& matrix, const StencilSpecT&... stencil_spec)
+  : HaloMatrixWrapper(matrix, GlobBoundSpec_t(), stencil_spec...) {}
+
+  HaloMatrixWrapper() = delete;
+
   ~HaloMatrixWrapper() {
     for(auto& dart_type : _dart_types) {
       dart_type_destroy(&dart_type);
     }
     _dart_types.clear();
   }
 
-  iterator begin() noexcept { return _begin; }
-
-  const_iterator begin() const noexcept { return _begin; }
-
-  iterator end() noexcept { return _end; }
-
-  const_iterator end() const noexcept { return _end; }
-
-  iterator_inner ibegin() noexcept { return _ibegin; }
-
-  const_iterator_inner ibegin() const noexcept { return _ibegin; }
-
-  iterator_inner iend() noexcept { return _iend; }
-
-  const_iterator_inner iend() const noexcept { return _iend; }
-
-  iterator_bnd bbegin() noexcept { return _bbegin; }
-
-  const_iterator_bnd bbegin() const noexcept { return _bbegin; }
-
-  iterator_bnd bend() noexcept { return _bend; }
-
-  const_iterator_bnd bend() const noexcept { return _bend; }
-
+  /// returns the underlying \ref HaloBlock
   const HaloBlock_t& halo_block() { return _haloblock; }
 
+  /**
+   * Initiates a blocking halo region update for all halo elements.
+   */
   void update() {
     for(auto& region : _region_data)
       update_halo_intern(region.second, false);
   }
 
+  /**
+   * Initiates a blocking halo region update for all halo elements within the
+   * the given region.
+   */
   void update_at(region_index_t index) {
     auto it_find = _region_data.find(index);
     if(it_find != _region_data.end())
       update_halo_intern(it_find->second, false);
   }
 
+  /**
+   * Initiates an asychronous halo region update for all halo elements.
+   */
   void update_async() {
     for(auto& region : _region_data)
       update_halo_intern(region.second, true);
   }
 
+  /**
+   * Initiates an asychronous halo region update for all halo elements within
+   * the given region.
+   */
   void update_async_at(region_index_t index) {
     auto it_find = _region_data.find(index);
     if(it_find != _region_data.end())
       update_halo_intern(it_find->second, true);
   }
 
+  /**
+   * Waits until all halo updates are finished. Only useful for asyncronous
+   * halo updates.
+   */
   void wait() {
     for(auto& region : _region_data)
       dart_wait_local(&region.second.halo_data.handle);
   }
 
+  /**
+   * Returns the local \ref ViewSpec
+   *
+   */
   const ViewSpec_t& view_local() const { return _view_local; }
 
-  const StencilSpecT& stencil_spec() const { return _stencil_spec; }
-
-  const HaloMemory_t& halo_memory() const { return _halomemory; }
+  /**
+   * Returns the halo memory management object \ref HaloMemory
+   */
+  HaloMemory_t& halo_memory() { return _halomemory; }
 
+  /**
+   * Returns the underlying NArray
+   */
   MatrixT& matrix() { return _matrix; }
```

Comment:
`const` versions of the two functions `matrix` and `halo_memory` would be nice

Reply:
done